gusto/time_discretisation.py

```diff
@@ -87,16 +87,32 @@ def __init__(self, domain, field_name=None, solver_parameters=None,
         self.courant_max = None
 
         if options is not None:
-            self.wrapper_name = options.name
-            if self.wrapper_name == "embedded_dg":
-                self.wrapper = EmbeddedDGWrapper(self, options)
-            elif self.wrapper_name == "recovered":
-                self.wrapper = RecoveryWrapper(self, options)
-            elif self.wrapper_name == "supg":
-                self.wrapper = SUPGWrapper(self, options)
+            if type(options) == MixedOptions:
+                self.wrapper = options
```

Comment:
Not that what you've done isn't a sensible way of implementing this, but it's occurred to me that it isn't quite consistent with how the existing wrappers work.

At the moment we have two broad types of objects: `Options` and `Wrappers`, and I think what we've done here is mix up the two a bit.

So now I'm thinking the best way of mirroring this structure for mixed function spaces would be to have:
1. `MixedFSOptions` which is defined in `configuration.py` and is essentially a glorified dictionary
2. `MixedFSWrapper` which is defined in `wrappers.py` and has the same methods as other `Wrapper` objects